### PR TITLE
provision accounts role

### LINKED ISSUE #16: ✏️ [TASK]: Add `provision_accounts` role
### Task Description

Add the [provision_accounts](https://github.com/UCL-MIRSG/Infrastructure/tree/main/ansible/playbooks/roles/provision_accounts) role. We'll need to generalise and rename the `os_mirsg_users` and `os_mirsg_users_groups` variables (which will also need to be renamed in [UCLMedicalImagingEnv](https://github.com/UCL-MIRSG/UCLMedicalImagingEnv/blob/main/environments/cs-production/group_vars/all/vars).

```[tasklist]
### Related Work Items
```

### PR DESCRIPTION
**Pull request by drmatthews**
Add a role for provisioning user accounts.

Closes #16

### CODE REVIEW

File: .github/workflows/molecule-provision_accounts.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Test provision
```
Comment (p-j-smith) [inline .github/workflows/molecule-provision_accounts.yml:1]:
```suggestion
name: Test provisioning accounts
```

File: .github/workflows/molecule-provision_accounts.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+on:
+  pull_request:
+    paths:
+      - "roles/provision/**"
```
Comment (p-j-smith) [inline .github/workflows/molecule-provision_accounts.yml:5]:
```suggestion
      - "roles/provision_accounts/**"
```

**Review by p-j-smith — CHANGES_REQUESTED** [review]
Nice! Just need to modify the workflow when the `provision_accounts` role changes instead of `provision`

File: .github/workflows/molecule-provision_accounts.yml (hunk 2 above)
Comment (drmatthews) [reply .github/workflows/molecule-provision_accounts.yml]: Whoops - thanks for picking that up!

Comment (p-j-smith) [reply .github/workflows/molecule-provision_accounts.yml:5]: no worries! It looks like these commits were undone somehow, perhaps when rebasing?

Comment (drmatthews) [reply .github/workflows/molecule-provision_accounts.yml]: Yup, exactly (d'oh). Put them back in now.

**Review by p-j-smith — APPROVED** [review]
looks good

**Pull request merged**